packages/fxa-auth-server/lib/routes/cloud-scheduler.ts

```diff
@@ -4,19 +4,96 @@
 
 import { ConfigType } from '../../config';
 import { AuthLogger, AuthRequest } from '../types';
-// commenting this out for now because the script uses p-queue, which is pure
-// ESM since 7.x.  That won't work when nodejs try to `require` it.
-// import { processDateRange } from '../../scripts/delete-unverified-accounts';
 import {
   AccountTasks,
   AccountTasksFactory,
-  //  ReasonForDeletion,
+  ReasonForDeletion,
 } from '@fxa/shared/cloud-tasks';
 import { StatsD } from 'hot-shots';
-import error from '../error';
+import { setupAccountDatabase } from '@fxa/shared/db/mysql/account';
 
 const MILLISECONDS_IN_A_DAY = 24 * 60 * 60 * 1000;
 
+/**
+ * Process a date range of accounts to delete.
+ *
+ * @param config
+ * @param accountTasks
+ * @param reason
+ * @param startDate
+ * @param endDate
+ * @param taskLimit
+ */
+export async function processDateRange(
+  config: ConfigType,
+  accountTasks: AccountTasks,
+  reason: ReasonForDeletion,
+  startDate: any,
+  endDate: any,
+  taskLimit: number
+) {
+  const PQueue = (await import('p-queue')) as any;
+  const kyselyDb = await setupAccountDatabase(config.database.mysql.auth);
+  const accounts = await kyselyDb
+    .selectFrom('accounts')
+    .where('accounts.emailVerified', '=', 0)
+    .where('accounts.createdAt', '>=', startDate)
+    .where('accounts.createdAt', '<=', endDate)
+    .leftJoin('accountCustomers', 'accounts.uid', 'accountCustomers.uid')
+    .select(['accounts.uid', 'accountCustomers.stripeCustomerId'])
+    .execute();
+
+  // Scaling suggestion is 500/5/50 rule, may start at 500/sec, and increase every 5 minutes by 50%.
+  // They also note increased latency may occur past 1000/sec, so we stop increasing as we approach that.
+  const scaleUpIntervalMins = 5;
+  let lastScaleUp = Date.now();
+  let rateLimit = taskLimit;
+
+  const queue = new PQueue({
+    interval: 1000,
+    intervalCap: rateLimit,
+    concurrency: rateLimit * 2,
+  });
+
+  if (accounts.length === 0) {
+    return 0;
+  }
+
+  for (const row of accounts) {
+    if (
+      rateLimit < 950 &&
+      Date.now() - lastScaleUp > scaleUpIntervalMins * 60 * 1000
+    ) {
+      rateLimit = Math.floor(rateLimit * 1.5);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      queue['#intervalCap'] = rateLimit; // This is private, but we need to update it
+      queue.concurrency = rateLimit * 2;
+      lastScaleUp = Date.now();
+    }
+
+    await queue.onSizeLessThan(rateLimit * 4); // Back-pressure
+
+    queue.add(async () => {
+      try {
+        const result = await accountTasks.deleteAccount({
+          uid: row.uid.toString('hex'),
+          customerId: row.stripeCustomerId || undefined,
+          reason,
+        });
+        console.log(
```

Comment:
console.log and console.error seemed more appropriate in a script.  Maybe change this to a log call?